ssl/t1_lib.c

```diff
@@ -788,7 +792,8 @@ static const uint16_t tls_default_sigalg[] = {
     TLSEXT_SIGALG_ecdsa_sha1, /* SSL_PKEY_ECC */
     TLSEXT_SIGALG_gostr34102001_gostr3411, /* SSL_PKEY_GOST01 */
     TLSEXT_SIGALG_gostr34102012_256_gostr34112012_256, /* SSL_PKEY_GOST12_256 */
-    TLSEXT_SIGALG_gostr34102012_512_gostr34112012_512 /* SSL_PKEY_GOST12_512 */
+    TLSEXT_SIGALG_gostr34102012_512_gostr34112012_512, /* SSL_PKEY_GOST12_512 */
+    0 /* SSL_PKEY_ED25519 */
```

Comment:
That table is the default signature algorithm associated with the certificate in the absence of a signature algorithms extension. That applies to TLS < 1.2 or TLS 1.2 if signature algorithms is not present only. It doesn't apply to TLS 1.3 because signature algorithms is mandatory.

Since Ed25519 can't be used with TLS < 1.3 it is zero.